app/src/main/kotlin/com/wire/android/ui/WireActivity.kt

```diff
@@ -153,13 +139,91 @@ class WireActivity : AppCompatActivity() {
     }
 
     @Composable
-    private fun handleDialogs() {
-        updateAppDialog(
-            { updateTheApp() },
-            viewModel.globalAppState.updateAppDialog
+    private fun isThereSharingIntent(): Boolean {
+        val fileSharingRestrictedDialogState = rememberVisibilityState<FileSharingRestrictedDialogState>()
+        FileSharingRestrictedDialogContent(
+            dialogState = fileSharingRestrictedDialogState
         )
+        if (featureFlagNotificationViewModel.featureFlagState.showFileSharingRestrictedDialog) {
+            fileSharingRestrictedDialogState.show(
+                fileSharingRestrictedDialogState.savedState ?: FileSharingRestrictedDialogState
+            )
+        }
+        if (featureFlagNotificationViewModel.featureFlagState.openImportMediaScreen) {
+            return true
+        }
+
+        return false
+    }
+
+    @Composable
+    fun setUpNavigationGraph(startDestination: String, navController: NavHostController, scope: CoroutineScope) {
+        Scaffold {
+            NavigationGraph(navController = navController, startDestination, viewModel.navigationArguments())
+        }
+        setUpNavigation(navController, scope)
+    }
+
+    @Composable
+    private fun setUpNavigation(
+        navController: NavHostController,
+        scope: CoroutineScope
+    ) {
+        val keyboardController = LocalSoftwareKeyboardController.current
+        // with the static key here we're sure that this effect wouldn't be canceled or restarted
+        LaunchedEffect("key") {
+            navigationManager.navigateState.onEach { command ->
+                if (command == null) return@onEach
+                keyboardController?.hide()
+                navController.navigateToItem(command)
+            }.launchIn(scope)
+
+            navigationManager.navigateBack.onEach {
+                if (!navController.popWithArguments(it)) finish()
+            }.launchIn(scope)
+
+            navController.addOnDestinationChangedListener { controller, _, _ ->
+                keyboardController?.hide()
+                updateScreenSettings(controller)
+            }
 
-        viewModel.globalAppState.conversationJoinedDialog?.let {
+            navController.addOnDestinationChangedListener(currentScreenManager)
+        }
+    }
+
+    @Composable
+    private fun handleDialogs() {
+        updateAppDialog({ updateTheApp() }, viewModel.globalAppState.updateAppDialog)
+        handleJoinConversationViaDeepLinkDialog(viewModel.globalAppState.conversationJoinedDialog)
+        handleCustomBackendDialog(viewModel.globalAppState.customBackendDialog.shouldShowDialog)
+        maxAccountDialog(viewModel::openProfile, viewModel::dismissMaxAccountDialog, viewModel.globalAppState.maxAccountDialog)
+        viewModel.globalAppState.blockUserUI?.let { AccountLoggedOutDialog(it, viewModel::navigateToNextAccountOrWelcome) }
```

Comment:
could we make it consistent with one of the namings ?

for example handleCustomBackendDialog -> customBackendDialog
handleJonConversationViewDeeplinkDialog -> conversationViewDeepLlinkDialog

also can we make : 
viewModel.globalAppState.blockUserUI?.let { AccountLoggedOutDialog(it, viewModel::navigateToNextAccountOrWelcome) } 

To follow the rest of dialogs

Reply:
Good idea.